### PR TITLE
Improve the AzDO MC Link

### PR DESCRIPTION
**Pull request by jonfortescue**
Improve the link to Mission Control that is outputted by the job waiter in the AzDO logs. This constructs the deep link.

### CODE REVIEW

File: src/Microsoft.DotNet.Helix/Sdk/HelixWait.cs [hunk 1]
```diff
@@ -70,11 +81,20 @@ private async Task<string> GetMissionControlResultUri()
...
+                }
+                catch (JsonException e)
+                {
+                    Log.LogMessage(MessageImportance.High, "Failed to parse JSON or find value in parsed JSON", e.StackTrace);
+                    return "Mission Control (generation of MC link failed -- JSON parsing error)";
                 }
-                string userName = JObject.Parse(githubJson)["login"].ToString();
 
-                return $"https://mc.dot.net/#/user/{userName}/builds";
+                return $"https://mc.dot.net/#/user/{userName}/{Source}/{Type}/{Build}";
```
Comment (akoeplinger) [inline src/Microsoft.DotNet.Helix/Sdk/HelixWait.cs]: there's still the issue that this won't work for PR builds as the user won't match the GitHub user represented by the token.

Comment (jonfortescue) [reply src/Microsoft.DotNet.Helix/Sdk/HelixWait.cs]: It will link directly to the guild's results, so it doesn't matter which token was used.

Comment (jonfortescue) [reply src/Microsoft.DotNet.Helix/Sdk/HelixWait.cs]: Oh, I think I see what you're saying. On line 89 we determine the Github username from the access token.

[the remark this answers is not in the document]
Comment (jonfortescue) [reply src/Microsoft.DotNet.Helix/Sdk/HelixWait.cs]: That's wild. I'll make that change!

[the remark this answers is not in the document]
Comment (jonfortescue) [reply src/Microsoft.DotNet.Helix/Sdk/HelixWait.cs]: https://github.com/dotnet/arcade/pull/1279

**Pull request merged**